custom_components/smartthinq_sensors/wideq/devices/ac.py

```diff
@@ -589,13 +595,23 @@ def is_mode_airclean_supported(self):
         """Return if AirClean mode is supported."""
         return self._is_mode_supported(SUPPORT_AIRCLEAN)
 
+    @cached_property
+    def is_mode_remove_moisture_supported(self):
+        """Return if RemoveMoisture mode is supported."""
+        return self._is_mode_supported(SUPPORT_REMOVE_MOISTURE)
+
+    @cached_property
+    def is_mode_power_save_supported(self):
+        """Return if PowerSave mode is supported."""
+        return self._is_mode_supported(SUPPORT_POWER_SAVE)
+
     @cached_property
     def supported_ligth_modes(self):
         """Return light switch modes supported."""
         if self._is_mode_supported(SUPPORT_LIGHT_SWITCH):
             return {MODE_OFF: LIGHT_DISPLAY_OFF, MODE_ON: LIGHT_DISPLAY_ON}
-        if self._is_mode_supported(SUPPORT_LIGHT_INV_SWITCH):
-            return {MODE_OFF: LIGHT_DISPLAY_INV_OFF, MODE_ON: LIGHT_DISPLAY_INV_ON}
+        # if self._is_mode_supported(SUPPORT_LIGHT_INV_SWITCH):
+        #     return {MODE_OFF: LIGHT_DISPLAY_INV_OFF, MODE_ON: LIGHT_DISPLAY_INV_ON}
```

Comment:
Why you comment this?

Reply:
Hi,
For me, RAC_88_DISPLAY_CONTROL displayed correctly the light on switch, because of the "inverted" code part, in my app the values were inverted, with that code commented, it displays fine.
Anyway, the commit https://github.com/ollo69/ha-smartthinq-sensors/pull/739/commits/d757db5e36d022f63fd3f3d67168d90b57651c00 was not part of the original pull request, after around 2 weeks I thought the pull request is not accepted so I used this fork to add some changes for my personal needs and use it on my HA.

The original pull request was only for commit https://github.com/ollo69/ha-smartthinq-sensors/pull/739/commits/4ee1db93944a204fd13fcfad13d54ca3053cf6fc1.

Thanks and regards,
Andrei

Reply:
> For me, RAC_88_DISPLAY_CONTROL displayed correctly the light on switch (not inverted as I saw others users reported). Because of the "inverted" code part, in my app the values were inverted. With that code commented, it displays fine for me.

This is the real problem in integration development, found the correct way to make it work properly for all possible devices. We should find a way to make this option work correctly for your device without breaking others.


> after around 2 weeks I thought the pull request is not accepted so I used this fork to add some changes for my personal needs and use it on my HA.

The only problem is that this year my time to work on this integration is very limited, but there are no reason to reject a PR without at least a feedback. If there are no answer just means that I had no time to analyze it, but PR are always welcome.

Please attach your device diagnostics so that I will also take a look to the display settings available to try to found a possible solution.

Reply:
Hi, 
Thanks for your reply. I have attached the device diagnostics, and looking into it I realised I remembered incorrectly the light problem. The "@RAC_88_DISPLAY_CONTROL" is not supported at all and "@BRIGHTNESS_CONTROL" that is used in integration's code as inverted value, for me it doesn't need to be inverted.
If you prefer, I could make a new pull request with only the Energy saving mode switch stuff, but not sure when, as I don't have a lot of time until the end of the month.

Thanks and regards,
Andrei

[smartthinq_sensors-26b2987ca676674af931188bf04abf81-AC Second Bedroom-09980ad75b0c1297483b0edf664cf99f.json](https://github.com/user-attachments/files/17360823/smartthinq_sensors-26b2987ca676674af931188bf04abf81-AC.Second.Bedroom-09980ad75b0c1297483b0edf664cf99f.json)

Reply:
I'm trying to figure out how to manage the light option more efficiently, as it seems that each device works differently (when it works at all)